src/PaymentGateways/PayPalCommerce/Actions/GetPayPalOrderFromRequest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Give\PaymentGateways\PayPalCommerce\Actions;
+
+use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\PaymentGateways\PayPalCommerce\Repositories\PayPalOrder;
+use PayPalHttp\HttpException;
+use PayPalHttp\IOException;
+
+/**
+ * This class returns paypal order for paypal order id given in request.
+ *
+ * @unreleased
+ */
+class GetPayPalOrderFromRequest
+{
+    /**
+     * @return \Give\PaymentGateways\PayPalCommerce\Models\PayPalOrder
+     * @throws HttpException | PaymentGatewayException | IOException
+     */
+    public function __invoke()
+    {
+        $paypalOrderId = give_clean($_POST['payPalOrderId']);
+
+        if ( ! $paypalOrderId) {
+            throw new PaymentGatewayException(
```

Comment:
I would suggest creating a custom exception here, which extends the `PaymentGatewayException`. Something like `PayPalOrderIdException`.

Reply:
fixed: https://github.com/impress-org/givewp/compare/3aa61ac2de8d...444927900f45